transaction-view/src/bytes.rs

```diff
@@ -0,0 +1,310 @@
+use crate::result::{Result, TransactionParsingError};
+
+/// Check that the buffer has at least `len` bytes remaining starting at
+/// `offset`. Returns Err if the buffer is too short.
+///
+/// Assumptions:
+/// - The current offset is not greater than `bytes.len()`.
+#[inline(always)]
```

Comment:
Why are you inlining all functions? Inlining a function does not guarantee better performance.

Reply:
because I tested and it does give better performance for my use-case

Reply:
In the follow-up PRs, I will post the difference with and without inlines.